vm/src/object/ext.rs

```diff
@@ -257,6 +269,88 @@ impl<T: PyObjectPayload> PyAtomicRef<T> {
     }
 }
 
+impl<T: PyObjectPayload> From<Option<PyRef<T>>> for PyAtomicRef<Option<T>> {
+    fn from(opt_ref: Option<PyRef<T>>) -> Self {
+        let val = opt_ref
+            .map(|x| PyRef::leak(x) as *const Py<T> as *mut i8)
+            .unwrap_or(null_mut());
+        Self {
+            inner: Radium::new(val),
+            _phantom: Default::default(),
+        }
+    }
+}
+
+impl<T: PyObjectPayload> PyAtomicRef<Option<T>> {
+    pub fn deref(&self) -> Option<&Py<T>> {
+        unsafe { self.inner.load(Ordering::Relaxed).cast::<Py<T>>().as_ref() }
+    }
+
+    /// # Safety
+    /// The caller is responsible to keep the returned PyRef alive
+    /// until no more reference can be used via PyAtomicRef::deref()
+    #[must_use]
+    pub unsafe fn swap(&self, opt_ref: Option<PyRef<T>>) -> Option<PyRef<T>> {
+        let val = opt_ref
+            .map(|x| PyRef::leak(x) as *const Py<T> as *mut i8)
+            .unwrap_or(null_mut());
+        let old = Radium::swap(&self.inner, val, Ordering::AcqRel);
+        unsafe { old.cast::<Py<T>>().as_ref().map(|x| PyRef::from_raw(x)) }
+    }
+
+    pub fn swap_to_temporary_refs(&self, opt_ref: Option<PyRef<T>>, vm: &VirtualMachine) {
+        let old = unsafe { self.swap(opt_ref) };
+        if let Some(old) = old {
+            if let Some(frame) = vm.current_frame() {
+                frame.temporary_refs.lock().push(old.into());
+            }
+        }
+    }
```

Comment:
a tiny style suggestion 
```suggestion
        let Some(old) = unsafe { self.swap(opt_ref) } else {
            return;
        };
        if let Some(frame) = vm.current_frame() {
            frame.temporary_refs.lock().push(old.into());
        }
    }
```